src/client/functions.ts

```diff
@@ -74,12 +74,40 @@ export const getPartyColor = (party: string) =>
 export const getMajorityRaceDisplay = (feature: DistrictGeoJSON) =>
   feature.properties.majorityRace && capitalizeFirstLetter(feature.properties.majorityRace);
 
-export function formatPvi(party?: Party, value?: number): string {
-  return value !== undefined && party !== undefined
-    ? `${party.label}+${Math.abs(value).toLocaleString(undefined, {
-        maximumFractionDigits: 0
-      })}`
-    : "N/A";
+/* Creates array of party-labelled pvi bucket counts as strings */
+export function formatPviByDistrict(
+  pviBuckets: readonly (PviBucket | undefined)[] | undefined
+): readonly string[] | undefined {
+  const partyLabels = ["R", "E (Even)", "D"];
+  // Count by partyLabels
+  const bucketCounts = pviBuckets?.reduce(
+    // eslint-disable-next-line
+    (allBuckets: { [key: string]: number } | undefined, bucket: PviBucket | undefined) => {
+      const name =
+        bucket &&
+        partyLabels.find(label => bucket.name.includes(label) || label.includes(bucket.name));
+      // eslint-disable-next-line
+      return name
+        ? allBuckets && name in allBuckets
+          ? { ...allBuckets, [name]: allBuckets[name] + 1 }
+          : { ...allBuckets, [name]: 1 }
+        : allBuckets;
+    },
+    {}
+  );
+  // Create string with partyLabels label
+  const bucketCountsStrings =
+    bucketCounts &&
+    partyLabels
+      .map((label: string) => {
+        return bucketCounts[label]
+          ? `${Math.abs(bucketCounts[label]).toLocaleString(undefined, {
```

Comment:
Is there a situation where `bucketCounts` can be negative? I would expect a count to be non-negative.
```suggestion
          ? `${bucketCounts[label].toLocaleString(undefined, {
```